pages/index.tsx

```diff
@@ -40,6 +40,11 @@ export default function Home({ results }: { results: MarkerData[] }) {
     []
   );
 
+  const memoizedMap = useMemo(
```

Comment:
`const memoizedMap = useMemo(
  () => <LeafletMap onClickMarker={toggleDrawer} data={results} />,
  [results]
);
`


First, the toggleDrawer is being invoked immediately inside the useMemo hook, which is not ideal. The onClickMarker prop should be passed a reference to the toggleDrawer function, not the result of calling it. This can be achieved by changing the implementation to onClickMarker={toggleDrawer}.